### PR TITLE
Add a private flag to indicate a thread is in SCC read mutex

### PR DESCRIPTION
**Pull request by hangshao0**
1. Add a private flag J9_PRIVATE_FLAGS2_IN_SHARED_CACHE_READ_MUTEX.
2. Add assertions on this private flag.

issue #9893

Signed-off-by: Hang Shao <hangshao@ca.ibm.com>

### CODE REVIEW

File: runtime/oti/j9consts.h [hunk 1]
```diff
@@ -174,7 +174,7 @@ extern "C" {
 #define J9_PRIVATE_FLAGS_STACKS_OUT_OF_SYNC 0x40000000
 #define J9_PRIVATE_FLAGS_FINAL_CALL_OUT_OF_MEMORY 0x80000000
 
-#define J9_PRIVATE_FLAGS2_UNUSED_0x1 0x1
+#define J9_PRIVATE_FLAGS2_IN_SHARED_CLASS_READ_MUTEX 0x1
```
Comment (pshipton) [inline runtime/oti/j9consts.h:177]: I think CACHE instead of CLASS is more accurate.

Comment (hangshao0) [reply runtime/oti/j9consts.h]: Fixed.

**Comment by pshipton** [comment]
jenkins test sanity,extended.system plinux jdk8

**Comment by pshipton** [comment]
Reverting until we have a fix for #9997

**Pull request merged**